src/Modules/UngradedModules/FreeResolutions.jl

```diff
@@ -204,28 +204,39 @@ function _extend_free_resolution(cc::Hecke.ComplexOfMorphisms, idx::Int)
   dom = domain(cc.maps[1])
   j   = 2
 
-  while j <= Singular.length(res)
+  #= get correct length of Singular sresolution =#
+  slen = iszero(res[Singular.length(res)+1]) ? Singular.length(res) : Singular.length(res)+1
```

Comment:
Unfortunately, this has not worked for me some days ago. I tried different examples and a fixed offset did not work out. Are you sure this does the job beyond the example provided by @wdecker ?

Reply:
According to @hannes14 this should be the way to go, I tried several examples and it worked. If you have some counterexample just let me know.

Reply:
Sorry, I didn't read it correctly. Probably this works.